File: Codeforces/1551/D1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, n, m, k;
    cin >> t;
    while (t--) {
        cin >> n >> m >> k;
        bool good = false;
        if (n % 2 == 1) {
            good = (k >= m / 2 and k % 2 == (m / 2) % 2);
        } else if (m % 2 == 1) {
            good = (k <= (m - 1) * n / 2 and k % 2 == 0);
        } else {
            good = (k % 2 == 0);
        }
        cout << (good ? "YES\n" : "NO\n");
    }
    return 0;
}

File: Codeforces/1551/D1.kt
private fun readLn() = readLine() ?: ""
private fun readLns() = readLn().split(" ")
private fun readInt() = readLn().toInt()
private fun readInts() = readLns().map { it.toInt() }

fun main() {
    val t = readInt()
    repeat(t) {
        val (n, m, k) = readInts().toList()
        val good = when {
            n % 2 == 1 -> (k >= m / 2 && k % 2 == (m / 2) % 2)
            m % 2 == 1 -> (k <= (m - 1) * n / 2 && k % 2 == 0)
            else -> (k % 2 == 0)
        }
        when (good) {
            true -> println("YES")
            false -> println("NO")
        }
    }
}